### PR TITLE
Add libp2p client connection

### PR DESCRIPTION
**Pull request by lrahmani**
## Proposed changes

- Add libp2p client connection.
- Update libp2p connection (and node) to act as delegate for client connection
- Add libp3p client unit tests

## Missing

- [x] Address ci 
- [x] Fix flaky unit test
- [ ] Update client connection yaml file to support multiple delegate nodes
- [ ] Test with aea cli
- [ ] Update client connection yaml file with public dht addresses as default delegate nodes
- [ ] Redeploy public DHT with new changes

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p/aea/api.go [hunk 1]
```diff
@@ -25,20 +25,22 @@ import (
...
+	host_public   string
+	port_public   uint16
+	host_delegate string
+	port_delegate uint16
+	msgin         *os.File
+	msgout        *os.File
+	out_queue     chan *Envelope
+	closing       bool
+	sandbox       bool
 }
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/aea/api.go:44]: Can we have comments somewhere here describing each attribute in more detail?

File: packages/fetchai/connections/p2p_libp2p/aea/api.go [hunk 2]
```diff
@@ -136,7 +144,7 @@ func (aea *AeaApi) Init() error {
 	if uri_public != "" {
 		parts = strings.SplitN(uri_public, ":", -1)
 		if len(parts) < 2 {
-			fmt.Println("[aea-api  ][error] malformed Uri:", uri)
+			fmt.Println("[aea-api  ][error] malformed Uri:", uri_public)
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/aea/api.go:147]: Was this change intentional? If so, comment should be `malformed Public URI:`

File: packages/fetchai/connections/p2p_libp2p/aea/api.go [hunk 3]
```diff
@@ -147,6 +155,21 @@ func (aea *AeaApi) Init() error {
 		aea.port_public = 0
 	}
 
+	// parse delegate uri
+	if uri_delegate != "" {
+		parts = strings.SplitN(uri_delegate, ":", -1)
+		if len(parts) < 2 {
+			fmt.Println("[aea-api  ][error] malformed Uri:", uri_delegate)
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/aea/api.go:162]: `malformed Delegate URI` or similar

File: packages/fetchai/connections/p2p_libp2p/aea/api.go (hunk 1 above)
Comment (DavidMinarsch) [reply packages/fetchai/connections/p2p_libp2p/aea/api.go:44]: You can reuse the ones from the python side

File: packages/fetchai/connections/p2p_libp2p_client/connection.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+            self._process_messages_task.cancel()
+            # TOFIX(LR) mypy issue https://github.com/python/mypy/issues/8546
+            # self._process_messages_task = None
+
+        logger.debug("disconnecting libp2p client connection...")
+        self._writer.write_eof()
+        await self._writer.drain()
+        self._writer.close()
+        # TOFIX(LR) requires python 3.7 minimum
+        # await self._writer.wait_closed()
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p_client/connection.py:195]: Can you change it to a python 3.6 compatible api please

File: packages/fetchai/connections/p2p_libp2p/aea/api.go (hunk 1 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/aea/api.go]: will refactor that

File: packages/fetchai/connections/p2p_libp2p/aea/api.go (hunk 2 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/aea/api.go:158]: it'll be redundant, as the uri will be printed.
there is a test in python class to catch user-input malformed uris

File: packages/fetchai/connections/p2p_libp2p_client/connection.py (hunk 4 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p_client/connection.py:195]: It is. Commented code requires python 3.7
